airflow/api_fastapi/core_api/routes/public/import_error.py

```diff
@@ -56,12 +82,34 @@ def get_import_error(
             status.HTTP_404_NOT_FOUND,
             f"The ImportError with import_error_id: `{import_error_id}` was not found",
         )
+    session.expunge(error)
+
+    auth_manager = get_auth_manager()
+    can_read_all_dags = auth_manager.is_authorized_dag(method="GET", user=user)
+    if not can_read_all_dags:
+        readable_dag_ids = auth_manager.get_permitted_dag_ids(user=user)
+        file_dag_ids = get_file_dag_ids(session, error.filename)
+
+        # Can the user read any DAGs in the file?
+        if not readable_dag_ids.intersection(file_dag_ids):
+            raise HTTPException(
+                status.HTTP_403_FORBIDDEN,
+                "You do not have read permission on any of the DAGs in the file",
+            )
+
+        # Check if user has read access to all the DAGs defined in the file
+        if not file_dag_ids.issubset(readable_dag_ids):
+            error.stacktrace = REDACTED_STACKTRACE
 
     return error
 
 
 @import_error_router.get(
     "",
+    dependencies=[
+        Depends(requires_access_view(AccessView.IMPORT_ERRORS)),
+        Depends(requires_access_dag("GET")),
```

Comment:
```suggestion
```
Same, you can access the endpoint if you don't have access to all dags. That's checked manually inside the body of the route.